client/state/analytics/middleware.js

```diff
@@ -18,6 +23,12 @@ const pageViewServices = {
 	default: ( { url, title } ) => analytics.pageView.record( url, title )
 };
 
+const loadTrackingTool = ( trackingTool ) => {
+	if ( trackingTool === 'Lucky Orange' ) {
+		analytics.luckyOrange.addLuckyOrangeScript();
+	}
+};
+
```

Comment:
You should use your `isTracking` selector here to determine if Lucky Orange is already on. To that end, you also need `state` to query from, so you should pass it in as an arg here.

Furthermore, you need to have [`analyticsMiddleware`](https://github.com/Automattic/wp-calypso/pull/12332/files#diff-e9d0416f7e9fe5479566ab065eb68f35R54) and `dispatcher` pass `state` down to `loadTrackingTool` (see below).